pkg/eval/unix/umask.go

```diff
@@ -0,0 +1,85 @@
+// +build !windows,!plan9,!js
+
+package unix
+
+import (
+	"errors"
+	"fmt"
+	"math"
+	"strconv"
+
+	"golang.org/x/sys/unix"
+
+	"github.com/elves/elvish/pkg/eval/vars"
+)
+
+//elvdoc:var umask
+//
+// The file mode creation mask. Its value is a string in Elvish octal
+// representation; e.g. 0o027. This makes it possible to use it in any context
+// that expects a `$number`.
+//
+// When assigning a new value a string is implicitly treated as an octal
+// number. If that fails the usual rules for interpreting
+// [numbers](./language.html#number) are used. The following arer equivalent:
+// `unix:umask = 027` and `unix:umask = 0o27`. You can also assign to it a
+// `float64` data type that has no fractional component.
+// The assigned value must be within the range [0 ... 0o777], otherwise the
+// assignment will throw an exception.
+//
+// You can do a temporary assignment to affect a single command; e.g.
+// `umask=077 touch a_file`. After the command completes the old umask will be
+// restored. **Warning**: Since the umask applies to the entire process, not
+// individual threads, changing it temporarily in this manner is dangerous if
+// you are doing anything in parallel. Such as via the
+// [`peach`](ref/builtin.html#peach) command.
+
+// UmaskVariable is a variable whose value always reflects the current file
+// creation permission mask. Setting it changes the current file creation
+// permission mask for the process (not an individual thread).
+type UmaskVariable struct{}
+
+var _ vars.Var = UmaskVariable{}
```

Comment:
The statement acts as an assertion that the UmaskVariable type satisfies the vars.Var interface.

Reply:
Thanks. Googling "go assert type satisfies interface" returns many documents that address this question. Here are a couple for those thinking about contributing pull-requests like this one in the future:

https://splice.com/blog/golang-verify-type-implements-interface-compile-time/
https://medium.com/@matryer/golang-tip-compile-time-checks-to-ensure-your-type-satisfies-an-interface-c167afed3aae